config/log_level.go

```diff
@@ -0,0 +1,109 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog"
+)
+
+// A LogLevel represents a logging level.
+type LogLevel string
+
+// Known log levels.
+const (
+	LogLevelUnset    LogLevel = ""
+	LogLevelTrace    LogLevel = "trace"
+	LogLevelDebug    LogLevel = "debug"
+	LogLevelInfo     LogLevel = "info"
+	LogLevelWarn     LogLevel = "warn"
+	LogLevelWarning  LogLevel = "warning"
+	LogLevelError    LogLevel = "error"
+	LogLevelCritical LogLevel = "critical"
+	LogLevelFatal    LogLevel = "fatal"
+	LogLevelPanic    LogLevel = "panic"
+	LogLevelOff      LogLevel = "off"
+	LogLevelNone     LogLevel = "none"
+	LogLevelDisabled LogLevel = "disabled"
+)
+
+// AllLogLevels are all of the known log levels.
+var AllLogLevels = [...]LogLevel{
+	LogLevelUnset,
+	LogLevelTrace,
+	LogLevelDebug,
+	LogLevelInfo,
+	LogLevelWarn,
+	LogLevelWarning,
+	LogLevelError,
+	LogLevelCritical,
+	LogLevelFatal,
+	LogLevelPanic,
+	LogLevelOff,
+	LogLevelNone,
+	LogLevelDisabled,
+}
+
+var logLevelLookup = func() map[LogLevel]struct{} {
+	m := map[LogLevel]struct{}{}
+	for _, lvl := range AllLogLevels {
+		m[lvl] = struct{}{}
+	}
+	return m
+}()
+
+// ValidateLogLevel validates that a log level is one of the known log levels.
+func ValidateLogLevel(lvl LogLevel) error {
+	_, ok := logLevelLookup[lvl]
+	if !ok {
+		return fmt.Errorf("unknown log level: %s", lvl)
+	}
+	return nil
+}
+
+// ToZerolog converts the log level to a level zerolog expects
```

Comment:
It might be nice to document here that "unset" will default to "info".

(totally optional)